### PR TITLE
Expand documentation

### PR DESCRIPTION
**Pull request by davkal**
- [x] Toplevel README
- [x] Getting started doc
- [x] Verify examples work
- [x] Usage (querying)
- [x] Contributing
- [x] Operations
- [x] Troubleshooting

### CODE REVIEW

File: production/README.md [hunk 1]
```diff
@@ -34,9 +35,16 @@ To test locally, we recommend using the docker-compose.yaml file in this directo
    docker-compose up
    ```
 
-1. Grafana should now be available at http://localhost:3000/.  Follow the [steps for configuring the datasource in Grafana](../docs/usage.md) and set the URL field to `http://loki:3100`.
+1. Grafana should now be available at http://localhost:3000/. Log in with `admin` / `admin` and follow the [steps for configuring the datasource in Grafana](../docs/usage.md), using `http://loki:3100` for the URL field.
 
-For instructions on how to use loki, see [our usage docs](../docs/usage.md).
+_Note_: When running locally, promtail starts before loki is ready. This can lead to the error message "Data source connected, but no labels received." After a couple seconds, Promtail will forward all newly created log messages correctly.
+Until this is fixed we recommend [building and running from source](#build-and-run-from-source).
```
[the remark this answers is not in the document]
Comment (davkal) [reply production/README.md]: https://github.com/grafana/loki/issues/267

File: README.md [hunk 2]
```diff
@@ -6,29 +6,37 @@
...
 
 ## Getting started
 
-For instructions on getting started with Loki, see [our getting started docs](./production/README.md).
+The [getting started docs](./production/README.md) have instructions on how to install Loki via Docker images, Helm charts, Jsonnet, or from source.
 
-For the beginnings of documentation on how to use Loki, see [our usage docs](./docs/usage.md).  [API documentation](./docs/api.md) is also available.
+Once you have promtail, Loki, and Grafana running, continue with [our usage docs](./docs/usage.md) on how to query your logs.
+
+### Advanced documentation
```
Comment (tomwilkie) [inline README.md]: Probably not so "advanced"?

File: README.md [hunk 3]
```diff
@@ -44,23 +52,18 @@ Your feedback is always welcome.
...
-- David Kaltschmidt's OSMC 2018 talk "[Logging is coming to Grafana](https://www.youtube.com/watch?v=vDXAVZY7rB0)"
-- David Kaltschmidt's KubeCon 2018 talk "[On the OSS Path to Full Observability with Grafana][kccna18-event]" ([slides][kccna18-slides], [video][kccna18-video])
-- Goutham Veeramachaneni's blog post "[Loki: Prometheus-inspired, open source logging for cloud natives](https://grafana.com/blog/2018/12/12/loki-prometheus-inspired-open-source-logging-for-cloud-natives/)".
+- David Kaltschmidt's KubeCon 2018 talk "[On the OSS Path to Full Observability with Grafana][kccna18-event]" ([slides][kccna18-slides], [video][kccna18-video]) on how Loki fits into a cloud-native environment.
+- Goutham Veeramachaneni's blog post "[Loki: Prometheus-inspired, open source logging for cloud natives](https://grafana.com/blog/2018/12/12/loki-prometheus-inspired-open-source-logging-for-cloud-natives/)" on details of the Loki architectire.
+- David Kaltschmidt's blog post "[Closer look at Grafana's user interface for Loki](https://grafana.com/blog/2019/01/02/closer-look-at-grafanas-user-interface-for-loki/)" on the ideas that went into the logging user interface.
```
Comment (tomwilkie) [inline README.md]: Mind adding link to https://speakerdeck.com/grafana/grafana-loki-like-prometheus-but-for-logs?

File: docs/grafana-provisioning.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+datasources:
+  - name: Loki
+    type: loki
+    access: proxy
+    url: http://localhost:3100
+    editable: false
+    basicAuth: true
+    basicAuthUser: my_user
+    basicAuthPassword: test_password
+```
```
Comment (tomwilkie) [inline docs/grafana-provisioning.md]: Where am I supposed to put these files?

File: docs/operations.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+2. Promtail reads the new line and increases its counters.
+3. Promtail forwards the line to a Loki distributor, its received counters should increase.
+4. The Loki distributor forwards it to a Loki ingester, its request duration counter increases.
+
+### Monitoring Mixins
+
+Check out our [Loki mixin](../production/loki-mixin) for a set of dashboards, recording rules, and alerts.
+These give you a comprehensive package on how to monitor Loki in production.
+
+For more information about mixins, take a look at the [Kubernetes mixins project](https://github.com/kubernetes-monitoring/kubernetes-mixin).
```
Comment (tomwilkie) [inline docs/operations.md]: Update link for https://github.com/monitoring-mixins/docs

File: docs/operations.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+By default, Loki stores everything on disk.
+The index is stored in a BoltDB under `/tmp/loki/index`.
+The chunks are stored under `/tmp/loki/chunks`.
+
+### Cloud storage
+
+Loki has support for Google Cloud storage.
+Take a look at our [production setup](https://github.com/grafana/loki/blob/a422f394bb4660c98f7d692e16c3cc28747b7abd/production/ksonnet/loki/config.libsonnet#L55) for the relevant configuration fields.
+
+Support for AWS and Azure is still under development.
```
Comment (tomwilkie) [inline docs/operations.md]: Its actually pretty possible right now, we just don't have docs for it.

**Comment by tomwilkie** [comment]
Merging without CLA as @davkal is a Grafana Labs employee.

**Pull request merged**